internal/metamorph/processor.go

```diff
@@ -345,8 +345,28 @@ func (p *Processor) StartSendStatusUpdate() {
 			case <-p.ctx.Done():
 				return
 
-			case message := <-p.statusMessageCh:
-				p.SendStatusForTransaction(message)
+			case msg := <-p.statusMessageCh:
+
+				// update status of transaction in storage
+				select {
+				case p.storageStatusUpdateCh <- store.UpdateStatus{
+					Hash:         *msg.Hash,
+					Status:       msg.Status,
+					Error:        msg.Err,
+					CompetingTxs: msg.CompetingTxs,
+				}:
+				default: // Ensure that writing to channel is non-blocking
+				}
+
```

Comment:
Aren't we going to lose data this way?

If the `p.storageStatusUpdateCh` will be full, then we will just simply NOT send the status there and lose that information.

Is this intentional?